testutil/promrated/prometheus.go

```diff
@@ -16,14 +16,54 @@
 package promrated
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"net/http"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 
 	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
 )
 
+const (
+	promEndpoint = "https://vm.monitoring.gcp.obol.tech"
+	promQuery    = "group%20by%20%28cluster_name%2C%20ccluster_hash%2C%20cluster_peer%2C%20cluster_network%2C%20pubkey_full%29%20%28core_scheduler_validator_balance_gwei%29"
+)
+
+type ClusterInfo struct {
+	ClusterName    string `json:"cluster_name"`
+	ClusterHash    string `json:"cluster_hash"`
+	ClusterNetwork string `json:"cluster_network"`
+	ClusterPeer    string `json:"cluster_peer"`
+}
+
+type PromMetric struct {
+	*ClusterInfo
+	PubKey string `json:"pubkey_full"`
+}
+
+type PromResult struct {
+	Metric PromMetric `json:"metric"`
+}
+
+type PromData struct {
```

Comment:
no need for all these types, you can just define a single nested type since it is only used for unmarshalling. Also, only include fields that you are actually using.
```
type promResponse struct {
  Data struct {
      Result []struct{
           ...
      } `json:result`
  }  `json:"data"`
}
```